backend/x11/backend.c

```diff
@@ -611,33 +617,9 @@ struct wlr_backend *wlr_x11_backend_create(struct wl_display *display,
 			wlr_log(WLR_ERROR, "Failed to query DRI3 DRM FD");
 			goto error_event;
 		}
-
-		char *drm_name = drmGetDeviceNameFromFd2(x11->drm_fd);
-		wlr_log(WLR_DEBUG, "Using DRM node %s", drm_name);
-		free(drm_name);
-
-		int drm_fd = fcntl(x11->drm_fd, F_DUPFD_CLOEXEC, 0);
-		if (drm_fd < 0) {
-			wlr_log(WLR_ERROR, "fcntl(F_DUPFD_CLOEXEC) failed");
-			goto error_event;
-		}
-
-		struct wlr_gbm_allocator *gbm_alloc = wlr_gbm_allocator_create(drm_fd);
-		if (gbm_alloc == NULL) {
-			wlr_log(WLR_ERROR, "Failed to create GBM allocator");
-			close(drm_fd);
-			goto error_event;
-		}
-		x11->allocator = &gbm_alloc->base;
 		pixmap_formats = &x11->dri3_formats;
 	} else if (x11->have_shm) {
```

Comment:
We can now support both DRI3 and SHM at the same time, so this can be an `if` instead of an `else if`.

Reply:
So we should store both pixman formats then right?

Reply:
Oh. Hm that won't work too well. We must use the `dri3_formats` if the allocator is GBM, `shm_formats` if the allocator is shm. However we don't know whether we'll use GBM or shm prior to `wlr_allocator_autocreate`. Potential solutions:

1. Support either DMA-BUFs or shm, not both at the same time. But this will break if DMA-BUFs are available (disabling shm) and the user forces the Pixman renderer (via `WLR_RENDERER=pixman`).
2. Check the type of the allocator returned by `wlr_allocator_autocreate`, with new functions like `allocator_is_gbm` and `allocator_is_shm`.
3. Do something more involved like `allocator_get_buffer_caps` then check the result to decide which formats to select, but let's keep this for later (if ever), it's over-complicated when there are no third-party allocators.